onnxruntime/core/providers/xnnpack/math/softmax.cc

```diff
@@ -166,24 +166,21 @@ Softmax::Softmax(const OpKernelInfo& info) : XnnpackKernel{info} {
   if (op_type_ == OpComputeType::op_compute_type_qu8) {
     // the order of input tensor, x,x_scale, x_zp, y_scale, y_zp
     OpQuantParam quant_param = ParseQuantParamForOp(info, x_dtype, 1);
-    xstatus = xnn_create_softmax_nc_qu8(channels,
-                                        channels,
-                                        channels,
-                                        quant_param[0].first[0],  // x_scale
-                                        quant_param[1].second,    // y_zp
-                                        quant_param[1].first[0],  // y_scale
-                                        0,                        // flags,
-                                        &p);
+    xstatus = xnn_create_softmax_nc_qu8(
+        quant_param[0].first[0],  // x_scale, input scale
+        quant_param[1].second,    // y_zp,  output zero point
+        quant_param[1].first[0],  // y_scale, output scale
+        0,                        // flags,
+        &p);
   } else if (op_type_ == OpComputeType::op_compute_type_fp32) {
-    xstatus = xnn_create_softmax_nc_f32(channels,
-                                        channels,
-                                        channels,
-                                        0,  // flags,
-                                        &p);
+    xstatus = xnn_create_softmax_nc_f32(
+        0,  // flags,
+        &p);
   }
 
-  ORT_ENFORCE(xstatus == xnn_status_success, "xnn_create_softmax_nc_",
+  ORT_ENFORCE(xstatus == xnn_status_success && p != nullptr, "xnn_create_softmax_nc_",
```

Comment:
Is there a scenario where it returns success but does not set `p`? Naively that seems like it would be fundamentally broken.

Reply:
I feel it's harmless to have an additional check. We cannot always trust an external library to work perfectly. The additional sanitize check is cheap.

Reply:
Yes it's harmless, but it feels like it's meaningless code of nearly no real value in this case as nothing is possible without `p`.

Error checking is generally a balance as to who does what. Duplicating the check in the caller and callee should not be necessary in our usage (vs. say software used by NASA). As the function in this case returns an explicit status, I would think it owns ensuring `p` is set if it returns success.

Reply:
removed it.